tests/eve_tests/unit_tests/test_pattern_matching.py

```diff
@@ -0,0 +1,81 @@
+import types
+from typing import Any, TypedDict
+
+import pytest
+
+from eve import Node
+from eve.pattern_matching import ModuleWrapper, ObjectPattern, get_differences
+
+
+class TestData(TypedDict):
+    name: str
+    a: Any
+    b: Any
+    expected_diff: list[tuple[str, str]]
+
+
+test_data: list[tuple[str, TestData]] = []
+
+
+def register_test_case(name, a, b, expected_diff):
+    test_data.append((name, {"a": a, "b": b, "expected_diff": expected_diff}))
```

Comment:
This is more convoluted than needed and not idiomatic (the TypedDict definition is also wrong: should not contain `name`). The simple way to implement the same workflow using regular `pytest` style would be something like:

```
# Each sample contains: (case_name, a, b, expected_diff)
test_data: list[tuple[str, Any, Any, list[tuple[str, str]]]] = [
    ("int_equal", 1, 1, [])
]

@pytest.mark.parametrize("name,a, b, expected_diff", test_data)
def test_all(name,a, b, expected_diff):
    diff = list(get_differences(a, b, path="a"))
    assert diff == expected_diff
```

Additionally, I miss some test cases dealing with nested data dict/lists/objects

Reply:
True, I've simplified as proposed and added more tests for nested things. Actually nesting was already tested before as the test for equality is already a recursive call.